themes/default/content/blog/api-gateway-to-eventbridge/index.md

```diff
@@ -0,0 +1,371 @@
+
+Add the following lines for the integration and route. The comments should explain what each block is doing:
+
+```typescript
+...
+
+// Define a policy granting API Gateway permission to publish to EventBridge.
+const apiGatewayRole = new aws.iam.Role("api-gateway-role",
+    {
+        assumeRolePolicy: {
+            Version: "2012-10-17",
+            Statement: [
+                {
+                    Action: "sts:AssumeRole",
+                    Effect: "Allow",
+                    Principal: {
+                        Service: "apigateway.amazonaws.com",
+                    },
+                },
+            ],
+        },
+        managedPolicyArns: [
+            "arn:aws:iam::aws:policy/AmazonEventBridgeFullAccess",
+        ],
+    },
+);
+
+// Create an API Gateway integration to forward requests to EventBridge.
+const integration = new aws.apigatewayv2.Integration("integration", {
+    apiId: api.id,
+
+    // The integration type and subtype.
+    integrationType: "AWS_PROXY",
+    integrationSubtype: "EventBridge-PutEvents",
+    credentialsArn: apiGatewayRole.arn,
+
+    // The body of the request to be sent to EventBridge. Note the
+    // event source matches pattern defined on the EventRule, and the
+    // Detail expression, which just forwards the body of the original
+    // API Gateway request (i.e., the uploaded document).
+    requestParameters: {
+        EventBusName: bus.name,
+        Source: "my-event-source",
+        DetailType: "my-detail-type",
+        Detail: "$request.body",
+    },
+});
+
+// Finally, define the route.
+const route = new aws.apigatewayv2.Route("route", {
+    apiId: api.id,
+    routeKey: "POST /uploads”,
+    target: pulumi.interpolate`integrations/${integration.id}`,
+});
+```
+
+With that, you’re ready to configure the Lambda.
+
+## Add a Lambda function handler
+
+The hard part’s done: you’ve declared an API and route, mapped that route to an integration, configured the integration to publish events to an event bus, and defined an event rule to respond to those events. All that's left now is to tell the rule _how_ to respond.
+
+So to finish things off, you need:
+
+* a Lambda function to handle uploads,
+* an EventBridge [_target_](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-targets.html) to bind your event rule to that function, and
+* an IAM policy granting EventBridge permission to invoke the function.
+
+Add the following lines to your program to complete it:
+
+```typescript
+...
```

Comment:
Generally, I find some people add this, too, and get confused. Do you end up finding the same thing in TypeScript? I've been commenting that out (e.g., `// ...`) to ensure the code at least compiles or runs in an interpreter.

Reply:
Yeah, I like this better --- it's a bit tedious to have to remove them I agree. We could also build a bit more cleverness into the [copy-button script](the copy handler) to exclude them when they're all that's on the first line, as I don't think they're meaningful in any language. 🤔